src/shared-components/button/components/linkButton/index.tsx

```diff
@@ -3,46 +3,59 @@ import PropTypes from 'prop-types';
 
 import { COLORS, COLORS_PROP_TYPES } from '../../../../constants';
 import Container from '../../shared-components/container';
+import { ButtonType } from '../..';
 import { ButtonContents, ButtonText } from '../../style';
 import { linkButtonStyles } from './style';
 
 const propTypes = {
-  disabled: PropTypes.bool,
-  children: PropTypes.node.isRequired,
+  as: PropTypes.oneOfType([PropTypes.string, PropTypes.elementType]),
+  buttonColor: COLORS_PROP_TYPES,
   buttonType: PropTypes.oneOf([
     'primary',
     'secondary',
     'tertiary',
     'quaternary',
   ]),
-  buttonColor: COLORS_PROP_TYPES,
-  as: PropTypes.oneOfType([PropTypes.string, PropTypes.elementType]),
+  children: PropTypes.node.isRequired,
+  disabled: PropTypes.bool,
   onClick: PropTypes.func,
   textColor: PropTypes.string,
 };
 
 const defaultProps = {
-  disabled: false,
-  buttonType: 'primary',
-  buttonColor: COLORS.primary,
   as: 'a',
+  buttonColor: COLORS.primary,
+  buttonType: 'primary',
+  disabled: false,
   onClick: () => undefined,
   textColor: '',
 };
 
+type LinkProps = {
+  as?: React.ReactNode;
```

Comment:
We can remove the TS-ignore if we replace this type with `as?: 'a' | 'span' | React.ElementType;`. (`ReactNode` is very permissive, and usually only appropriate for things like `children`)

It appears that, with the types, we're able to infer that `'a'` and `'span'` are used as `React.ElementType`. Adding `'a' | 'span'` is helpful, I think, for the API on the consuming end, though.

Reply:
I am OK with this change, I knew about `React.ElementType` would solve the issue but wouldnt allow the strings. But we are not going to allow any other html tag. Although I think we will only ever use this for `a` and `Link` comp.
Note that I will have to update the docs accordingly

Reply:
Yep--that sounds right. Looks like the `as={clickable ? 'a' : 'span'}` I saw was for some other component. Sorry about that!

Reply:
I changed it to `string | React.ElementType;` (I thought I tried this before and didn't work without the tsignore)

Reply:
Any reason we shouldn't limit the specific string type to `'a'`? I'd prefer errors be raised dev side immediately, from typing, than pass something in that we didn't really build the component for.

Reply:
By the component doc I thought it accepted any html element but we don't have any other case than "a" so we'll do `"a" | React.ElementType;` for now since is the stricter option

Reply:
Sorry if I wasn't clear about this! I'm mostly suggesting  we should take advantage of adding TypeScript types to add specificity to our usage. 🙂